source/client/stream_decoder.h

```diff
@@ -78,8 +120,40 @@ class StreamDecoder : public Envoy::Http::StreamDecoder,
   const Envoy::MonotonicTime connect_start_;
   Envoy::MonotonicTime request_start_;
   bool complete_;
-  bool measure_latencies_;
+  bool measure_latencies_{true};
   const uint32_t request_body_size_;
+  Nighthawk::PoolImpl<PoolableStreamDecoder>::PoolablePtr self_;
+  MilestoneTrackerImpl milestone_tracker_;
+  uint32_t milestone_request_start_;   // in queue or connecting
+  uint32_t milestone_pool_ready_;      // connection ready to start request
+  uint32_t milestone_response_header_; // received response header
+  uint32_t milestone_first_body_byte_; // received the first body byte
+  uint32_t milestone_last_body_byte_;
+  uint32_t milestone_trailer_;
+  uint32_t milestone_response_complete_;
+  bool have_first_body_byte_{false};
+};
+
+class PoolableStreamDecoder : public Client::StreamDecoder, public PoolableImpl {
+public:
+  PoolableStreamDecoder(Envoy::TimeSource& time_source,
+                        Client::StreamDecoderCompletionCallback& decoder_completion_callback,
+                        std::function<void()> caller_completion_callback,
+                        Statistic& connect_statistic, Statistic& latency_statistic,
+                        const Envoy::Http::HeaderMap& request_headers, bool measure_latencies,
+                        uint32_t request_body_size)
+      : StreamDecoder(time_source, decoder_completion_callback,
+                      std::move(caller_completion_callback), connect_statistic, latency_statistic,
+                      request_headers, measure_latencies, request_body_size) {}
+};
+
+class StreamDecoderPoolImpl : public PoolImpl<PoolableStreamDecoder> {
```

Comment:
Does this need to be a class? I think it could just be a `using` alias or a `typedef`.

Reply:
Oh, yes, much nicer, done.